### PR TITLE
Remove void as possible option type

### PR DESCRIPTION
**Pull request by nafur**
This PR removes the possibility to have `void` options. While there are (very) few options that may legitimately be considered `void` (they do not actually hold a value, but merely trigger a handler or predicate), the type being `void` introduces a number of special cases in the options setup. Their only real benefit was to avoid having data members that would never be used.
As it turns out, though, the same can be achieved by not specifying a `name`, which already supported as well.

### CODE REVIEW

File: src/options/base_options.toml [hunk 1]
```diff
@@ -63,16 +63,18 @@ name   = "Base"
   category   = "common"
   short      = "v"
   long       = "verbose"
-  type       = "void"
-  handler    = "increaseVerbosity"
+  type       = "bool"
```
Comment (4tXJ7f) [inline src/options/base_options.toml:66]: If we declare those options to be of `type` `bool`, don't we also start accepting `--no-verbose`?

Comment (nafur) [reply src/options/base_options.toml:66]: note the `alternate = false` below. This prevents exactly that.

Comment (4tXJ7f) [reply src/options/base_options.toml:66]: Ah, great!

**Review by 4tXJ7f — APPROVED** [review]
LGTM, thanks for the explanation.

**Pull request merged**